### DISCUSSION TITLE
Keep the session alive for whatsapp Web

### DISCUSSION THREAD

**Discussion by dheerain** [jeliebig/WAWebSessionHandler#11] [Q&A]
My Whatsapp web session gets disconnected when stays idle for few minutes. How to keep it active all the time .?

**Comment by TSRBerry** [comment]
Could you explain a bit further what happens? I am not sure if I understand what you mean by "stays idle".

**Reply by dheerain** [reply]
When there are no new incoming message and whatsapp web is kept in the background and no input by the user on the browser for long time . Then session gets disconnected

**Reply by TSRBerry** [reply]
Oh okay, I'll look into it in a few days. I hope I can do something to prevent that.

Do I need to minimize the window for it to happen? Or does it happen if I just don't use the window?

**Reply by dheerain** [reply]
No Just keep it in the background . I am not sure if it is with my system only.
Thanks for checking.

 Once more question Since today morning I am getting another issue where the whatsapp session is not getting loaded from file and asking me to scan the QR code again to login every time I run the program

**Reply by TSRBerry** [reply]
Okay, thank you!

Oh that sucks, I'll look into it. They might have updated something again. Your file should still be working as soon as I update my script.

Edit:
They really have updated it, but my script still works for me. Are you using the latest version? If so, can you provide the output of the console window, so I can investigate?

**Reply by dheerain** [reply]
It is random..!!! Sometimes it is able to pick up the session but sometimes it fails . I will keep a check

**Reply by dheerain** [reply]
How I can I verify that the Script has loaded the whatsapp session successfully.?

**Reply by TSRBerry** [reply]
the script itself can't really verify (atm) that it has loaded the session successfully, but you can verify that it is done loading the session if you see something like this in your console:

> 2021-05-22 18:55:30,141 [DEBUG] (access_by_file): WaSession object is valid.
> 2021-05-22 18:55:30,141 [DEBUG] (__start_session): Starting browser... [HEADLESS: False]
> 2021-05-22 18:55:36,934 [DEBUG] (__start_session): Loading WhatsApp Web...
> 2021-05-22 18:55:38,145 [DEBUG] (__set_indexed_db_user): Inserting setIDBObjects function...
> 2021-05-22 18:55:38,258 [DEBUG] (__set_indexed_db_user): setIDBObjects function inserted.
> 2021-05-22 18:55:38,258 [DEBUG] (__set_indexed_db_user): Writing IDB data...
> 2021-05-22 18:55:38,284 [DEBUG] (__set_indexed_db_user): Waiting until all objects are written to IDB...
> 2021-05-22 18:55:39,386 [DEBUG] (access_by_obj): Reloading WhatsApp Web...
> 2021-05-22 18:55:40,260 [DEBUG] (access_by_obj): Waiting until the browser window is closed...

**Reply by dheerain** [reply]
I am not sure what is the issue..!! Session file loads for only one run . For the next run It fails to load the session file and I need to scan the QR code manually

**Reply by TSRBerry** [reply]
That actually made it a lot easier to get an idea where the issue is coming from, thank you!

I need to look into that again and check if this is something that I can fix with my script. I'll respond again as soon as I have more information.

**Reply by dheerain** [reply]
Thanks

**Reply by dheerain** [reply]
Do you have any update..?

**Reply by TSRBerry** [reply]
Hello,
Sorry for the late update. The issue (#12) should be fixed with the latest commit. Please update your version of the script and look if you experience any issues.

For keeping the session alive I might need a bit more time, as I am not quite sure how it detects that you are active on the page.

**Reply by TSRBerry** [reply]
How long do I need to keep the window open, to disconnect the session?

**Comment by dheerain** [comment]
Hi,
I recently joined whatsapp web beta and since then I am not able to save the session ..

Logs Shows like that. it keeps on waiting 

12:29:30,957 [DEBUG] (__get_indexed_db_user): Executing getIDBObjects function...
2021-07-17 12:29:30,965 [DEBUG] (__get_indexed_db_user): Waiting until IDB operation finished...
2021-07-17 12:29:30,968 [DEBUG] (__get_indexed_db_user): Getting IDB results...
2021-07-17 12:29:31,987 [DEBUG] (__get_indexed_db_user): Executing getIDBObjects function...
2021-07-17 12:29:31,996 [DEBUG] (__get_indexed_db_user): Waiting until IDB operation finished...
2021-07-17 12:29:31,998 [DEBUG] (__get_indexed_db_user): Getting IDB results...
2021-07-17 12:29:33,015 [DEBUG] (__get_indexed_db_user): Executing getIDBObjects function...


Can you help me fix that

**Reply by TSRBerry** [reply]
Hi,
I will look into this and see what I can do for you. Since I am not part of WhatsApp web beta this might be a little harder, but I'll get back to you when I have more information about your issue.

**Reply by TSRBerry** [reply]
Everything seems to be working fine for me, so I need to ask you a few things.
Are you using the most recent version of my script?
Could you post the complete console output of my script?
